lib/project_types/extension/models/specification_handlers/checkout_argo_extension.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Extension
+  module Models
+    module SpecificationHandlers
+      class CheckoutArgoExtension < Default
```

Comment:
Will all checkout extensions use this as their `Specification#identifier`? If so, we should possibly consider just calling it a checkout extension and leave Argo out of it. If this doesn't map to the specification identifier, we should follow the naming and mechanism that's currently being introduced in https://github.com/Shopify/shopify-app-cli/pull/1230/files, which provides another layer of fallback before opting to use `SpecificationHandlers::Default`.

Reply:
Most new checkout argo extensions will use this except for post purchase. The later will still use its own handler for a while still and it only allows for `metafields` as a key in the config files. I'm all for making a higher level `CheckoutExtension` class and each type of extension can define their own permitted keys. However, do you know when we expect this PR to be shipped? We are kind of tight timeline for the Unite demos which include this patch 😅